cmp/grid/GridModel.js

```diff
@@ -1014,6 +1019,23 @@ export class GridModel extends HoistModel {
         }
     }
 
+    /**
+     * Returns true as soon as the underlying agGridModel is ready, waiting a limited period
+     * of time if needed to allow the component to initialize. Returns false if grid not ready
+     * by end of timeout to ensure caller does not wait forever (if e.g. grid is not mounted).
+     * @param {number} [timeout] - timeout in ms
+     * @return {Promise<boolean>} - latest ready state of grid
+     */
+    async getIsReadyAsync(timeout = 3 * SECONDS) {
```

Comment:
wonder if this could be called `whenReadyAsync()` instead.  Seems to be slightly clearer -- current name seems like a conventional getter around the flag.

Reply:
Yes, I went back and forth on the name. It felt a bit weird to have `whenReadyAsync` return an `isReady` value that might be false, but not that weird. I'm a-OK with where it landed - I don't expect this to be a common external API in any case - was very much focusing on a clean way to resolve the edge-but-bad case in the ticket. Thanks for merging!